foxsec_sqs_worker.py

```diff
@@ -0,0 +1,231 @@
+    return decorator
+
+@retry(retry_count=5, delay=5)
+def waf_update_ip_set(waf_ipset_id, waf_updates):
+    """Update WAF ip set"""
+    # Get our change token
+    try:
+        change_token = WAFREGIONAL.get_change_token()  # print(token['ChangeToken'])
+    except ClientError as err:
+        print("waf get_change_token failed: %s" % err)
+        return False
+
+    # ChangeTokenStatus, ResponseMetadata.RetryAttempts
+    try:
+        token_status = WAFREGIONAL.get_change_token_status(ChangeToken=change_token['ChangeToken'])
+        print("TOKEN %s" % change_token['ChangeToken'])
+        pprint(token_status)
+    except ClientError as err:
+        print("waf get_change_token_status failed: %s" % err)
+        return False
+
+    # Update our WAF ipset and return if successful
+    try:
+        WAFREGIONAL.update_ip_set(IPSetId=waf_ipset_id,
+                                  ChangeToken=change_token['ChangeToken'],
+                                  Updates=waf_updates)
+    except ClientError as err:
+        print("waf update_ip_set failed: %s" % err)
+        return False
+    else:
+        return True
+
+
+def sqs_delete_messages(arn, messages):
+    """Delete messages from SQS arn"""
+    # Parse SQS ARN
+    arn = parse_arn(arn)
+
+    # Get queue url from queue name
+    try:
+        queue_url = SQS.get_queue_url(QueueName=arn.get('resource'))
+    except ClientError as err:
+        print("sqs get_queue_url failed: %s" % err)
+        return False
+
+    # Delete received messages and return if successful
+    try:
+        SQS.delete_message_batch(QueueUrl=queue_url.get('QueueUrl'), Entries=messages)
+    except ClientError as err:
+        print("sqs delete_message_batch failed: %s" % err)
+        return False
+    else:
+        return True
+
+
+def ip_address_validate(address):
+    """Confirm valid IP and identify v4/v6"""
+    ip_types = {4: 'IPV4', 6: 'IPV6'}
+
+    try:
+        ip_network = ipaddress.ip_network(address)
+    except ValueError as err:
+        print("ip_network failed: %s" % err)
+        return err
+    else:
+        return str(ip_network), ip_types[ip_network.version]
+
+
+def dynamodb_put_items(items):
+    """Put items in DynamoDB. One item at a time."""
+    for item in items:
+        try:
+            DYNAMODB.put_item(Item=item)
+        except ClientError as err:
+            print("dybamodb put item failed: %s" % err)
+
+
+def lambda_handler(event, context):
+    """Main entrypoint handler"""
+    # Initialize
+    del context  # Unused
+    waf_updates = []
+    sqs_entries = []
+    dynamodb_items = []
+
+    # Evaluate each record
+    record = None
+    for record in event.get('Records'):
+        # Mark SQS message for deletion early
+        sqs_entry = {
+            'Id': record.get('messageId'), 
+            'ReceiptHandle': record.get('receiptHandle')
+        }
+        sqs_entries.append(sqs_entry)
+
+        # Skip invalid records
+        if not record.get('body'):
```

Comment:
`if not` is a bit nebulous in that it can hit None, False, etc. I would recommend trying to be more specific if possible, e.g. `if record.get('body') is None`.